pages/restaurants/index.js

```diff
@@ -1,24 +1,82 @@
 import RestaurantsList from "../../components/RestaurantsList";
 import Heading from "../../components/Heading";
 import { StyledMain } from "../../components/styles";
+import { AlertMessage } from "../../components/styles";
+import { ChevronsDownIcon } from "../../public/icons";
+import styled from "styled-components";
 import useSWR from "swr";
+import { useState } from "react";
+import { useEffect } from "react";
 
-export default function Restaurants({ onToggleFavorite, username, onLogin }) {
-  const { data: restaurants } = useSWR("/api/restaurants", {
-    fallbackData: [],
-  });
+export default function Restaurants({
+  onToggleFavorite,
+  username,
+  onLogin,
+  restaurantsMatchingTheSearch,
+}) {
+  const [currentFilterCondition, setCurrentFilterCondition] =
+    useState("Ohne Auswahl");
+  const [isShowFilterConditions, setIsShowFilterConditions] = useState(false);
+  const [matchedRestaurants, setMatchedRestaurants] = useState(
+    restaurantsMatchingTheSearch
+  );
   const { data: userInfos } = useSWR("/api/user-infos", {
     fallbackData: [],
   });
 
+  const filterConditions = [
+    { german: "Bewertung (höchste zuerst)", english: "rating" },
+    { german: "Küche", english: "cuisine" },
+    { german: "Preisniveau(teuerste zuerst)", english: "priceLevel" },
+  ];
+
+  function handleOpenFilterList() {
+    setIsShowFilterConditions(!isShowFilterConditions);
+  }
+
+  function handleFilterCondition(condition) {
+    function compareFunction(a, b) {
+      if (a[condition.english] > b[condition.english]) return -1;
+      if (a[condition.english] < b[condition.english]) return 1;
+      return 0;
+    }
```

Comment:
I guess you know that there is no need for this extra function to be called three lines below; have you done it just out of readability issues?

Reply:
Yes. That's what I considered before.  I will put it back as call back function.